util/json.go

```diff
@@ -21,7 +29,16 @@ import (
 func UnmarshalJSON(bs []byte, x interface{}) (err error) {
 	buf := bytes.NewBuffer(bs)
 	decoder := NewJSONDecoder(buf)
-	return decoder.Decode(x)
+	if err := decoder.Decode(x); err != nil {
+		return err
+	}
+
+	// Since decoder.Decode validates only the first json structure in bytes,
+	// check if decoder has more bytes to consume to validate whole input bytes.
+	if decoder.More() {
```

Comment:
Looking a little deeper at the implementation for `decoder.More` I think what we actually want is to use `decoder.Token`, so something more like:

```go
	if _, err := decoder.Token(); err != io.EOF {
		return fmt.Errorf("invalid JSON at offset %d, expected EOF", decoder.InputOffset())
	}
```

Reply:
Thanks for the review.
I agree with you, and I fixed to use `decoder.Token` to validate.